From c0e48edacc3c12e209857c24366cff7665fe1916 Mon Sep 17 00:00:00 2001
From: Maximilian Seesslen <maximilian.seesslen@ifm.com>
Date: Wed, 14 Mar 2018 18:31:08 +0100
Subject: [PATCH 1/2] ex-timestamp: replace put_time

put_time is part of c++11 put was not implemented in gcc 4.9
---
 ex-timestamp.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/ex-timestamp.cpp b/ex-timestamp.cpp
index 998cb79..ce0e087 100644
--- a/ex-timestamp.cpp
+++ b/ex-timestamp.cpp
@@ -37,9 +37,12 @@ std::string formatTimestamp(ifm3d::TimePointT timestamp) {
           timestamp.time_since_epoch() - duration_cast<seconds>(timestamp.time_since_epoch()));
 
     std::ostringstream s;
-
-    s << std::put_time(std::localtime(&time), "%Y-%m-%d %H:%M:%S")
-      << ":" << std::setw(3) << std::setfill('0') << milli.count();
+    std::tm *lt=std::localtime(&time)
+    // put_time not supported in gcc 4.9
+    //s << std::put_time(std::localtime(&time), "%Y-%m-%d %H:%M:%S")
+    s << lt->tm_year << lt->tm_mon << lt->tm_mday << " "
+        << lt->tm_hour << lt->tm_min << lt->tm_sec
+        << ":" << std::setw(3) << std::setfill('0') << milli.count();
 
     return s.str();
 }
-- 
2.7.4

